frontend/app/components/redesign/components/LinkTagGenerator.tsx

```diff
@@ -0,0 +1,153 @@
+import { useState } from 'react'
+import { cx } from 'class-variance-authority'
+import { Heading5 } from '../Typography'
+import { InputField, ToolsPrimaryButton } from './index'
+import { SVGCopyIcon } from '~/assets/svg'
+
+export const LinkTagGenerator = () => {
+  const [pointerInput, setPointerInput] = useState('')
+  const [linkTag, setParsedLinkTag] = useState('')
+  const [invalidUrl, setInvalidUrl] = useState(false)
+  const [showCodeBox, setShowCodeBox] = useState(false)
+  const [isCopied, setIsCopied] = useState(false)
+
+  // Function to validate and format the payment pointer URL
+  const isValidPointer = (input: string) => {
+    try {
+      let urlString = input.trim()
+      if (input.charAt(0) === '$') {
+        urlString = input.replace('$', 'https://')
+      }
+
+      const url = new URL(urlString)
+
+      if (url.pathname === '/') {
+        return `${url.origin}.well-known/pay`
+      }
+
+      return url.href
```

Comment:
Should exclude anything other than origin + pathname from the result. WM doesn't care about searchparams and hash.